src/zcl-packet/lib/foundation.ts

```diff
@@ -624,14 +624,8 @@ ru.clause('uint64', function (name) {
 ru.clause('strPreLenUint8', function (name) {
     parsedBufLen += 1;
     this.uint8('len').tap(function () {
-        var attrId = this.vars['attrId'];
-        // special xiaomi struct-string
-        if (attrId === 65281) {
-            ru['xiaoMiStruct'](name)(this);
```

Comment:
Instead of moving it to zigbee-shepherd-converters, couldn't we just simply add a try catch here? If decoding fails (for eg the doorlock) we use the default code.

Reply:
@Koenkk nice ideas, i will try to catch it.